### PR TITLE
Fix(playwright): amend build config

### PR DESCRIPTION
**Pull request by aizlewood40**

### CODE REVIEW

File: packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts [hunk 1]
```diff
@@ -27,6 +27,12 @@ export function updateTasksYaml(
                           'npx nx affected --base="$BASE_SHA" --target=e2e --parallel=1',
                       ],
                   },
+                  'e2e:updatesnapshots': {
+                      description: 'Run e2e snapshot updates in ci',
+                      command: [
+                          'npx nx affected --base="$BASE_SHA" --target=e2e --parallel=1 --skip-nx-cache --update-snapshots --grep @visual-regression',
+                      ],
+                  },
```
Comment (Danphillipz) [inline packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts:35]: Just confirming, can't quite tell, but is this only added if visual regression == native?

File: packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts [hunk 2]
```diff
@@ -51,21 +57,8 @@ export function updateTaskctlYaml(
...
-            taskctl.pipelines.prod.push({ task: 'e2e:ci' });
-        } else {
-            // Add tasks
-            taskctl.pipelines.dev.push({ task: 'e2e' });
-            taskctl.pipelines.fe.push({ task: 'e2e' });
-            taskctl.pipelines.nonprod.push({ task: 'e2e' });
-            taskctl.pipelines.prod.push({ task: 'e2e' });
-        }
+    if (taskctl.pipelines && visualRegression) {
+        taskctl.pipelines.updatesnapshots = [{ task: 'e2e:updatesnapshots' }];
```
Comment (Danphillipz) [inline packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts:61]: Also confirming here, but is this only added if visual regression == native?

**Review by Danphillipz — COMMENTED** [review]
Great effort on this mate

File: packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts (hunk 2 above)
Comment (aizlewood40) [reply packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts:61]: it's only called when native

File: packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts (hunk 1 above)
Comment (aizlewood40) [reply packages/playwright/src/generators/visualRegression/utils/update-tasks-yamls.ts:35]: Only called when native

File: packages/playwright/src/generators/init/generator.spec.ts [hunk 3]
```diff
@@ -168,11 +202,134 @@ describe('playwright generator', () => {
...
             appTree,
             joinPathFragments(projectNameE2E, 'project.json'),
         );
         expect(projectJson.targets.e2e).toBeTruthy();
     }, 100_000);
+
+    it('should alter azure devops stages file that is generated by workspace', async () => {
+        appTree.write(
+            'build/azDevOps/azuredevops-stages.yaml',
+            `parameters:
```
Comment (nandorszentpeteri) [inline packages/playwright/src/generators/init/generator.spec.ts:234]: Please move this line to the next one so the indentation remains valid for a YAML file.

File: packages/playwright/src/generators/init/generator.ts [hunk 4]
```diff
@@ -57,7 +59,7 @@ function updateDependencies(tree) {
         tree,
         {},
         {
-            playwright: '*',
+            playwright: '^1.31.1',
```
Comment (nandorszentpeteri) [inline packages/playwright/src/generators/init/generator.ts:62]: This version should come for a const, like in the other generators.

**Review by Danphillipz — APPROVED** [review]

**Pull request merged**